src/ControlSystem/ControlErrors/Size.hpp

```diff
@@ -0,0 +1,298 @@
+ *
+ * In addition to calculating the control error, if the
+ * `control_system::Tags::WriteDataToDisk` tag inside the
+ * `Parallel::GlobalCache` is true, then a diagnostic file named
+ * `Diagnostics.dat` is also written to the same group that
+ * `control_system::write_components_to_disk` would write the standard control
+ * system output (`/ControlSystems/Size/`). The columns of this diagnostic file
+ * are as follows (with a small explanation if the name isn't clear):
+ *
+ * - %Time
+ * - ControlError
+ * - StateNumber: Result of `control_system::size::State::number()`
+ * - DiscontinuousChangeHasOccurred: 1.0 for true, 0.0 for false.
+ * - FunctionOfTime
+ * - dtFunctionOfTime
+ * - HorizonCoef00
+ * - dtHorizonCoef00
+ * - MinDeltaR: The minimum of the `StrahlkorperGr::radial_distance` between the
+ *   horizon and the excision surfaces.
+ * - MinRelativeDeltaR: MinDeltaR divided by the `Strahlkorper::average_radius`
+ *   of the horizon
+ * - ControlErrorDeltaR: \f$ \dot{S}_{00} (\lambda_{00} -
+ * r_{\mathrm{excision}}^{\mathrm{grid}} / Y_{00}) / S_{00} - \dot{\lambda}_{00}
+ * \f$
+ * - TargetCharSpeed
+ * - MinCharSpeed
+ * - MinComovingCharSpeed: Eq. 98 in ArXiv:1211.6079
+ * - CharSpeedCrossingTime: %Time at which the min char speed is predicted to
+ *   cross zero and become negative.
+ * - ComovingCharSpeedCrossingTime: %Time at which the min comoving char speed
+ * is predicted to cross zero and become negative.
+ * - DeltaRCrossingTime: %Time at which the distance between the excision and
+ *   horizon surfaces is predicted to be zero.
+ * - SuggestedTimescale: A timescale for the `TimescaleTuner` suggested by one
+ * of the State%s
+ * - DampingTime
+ */
+template <::domain::ObjectLabel Horizon>
+struct Size : tt::ConformsTo<protocols::ControlError> {
+  static constexpr size_t expected_number_of_excisions = 1;
+
+  using object_centers = domain::object_list<Horizon>;
+
+  struct MaxNumTimesForZeroCrossingPredictor {
+    // Int so we get proper bounds checking
+    using type = int;
+    static constexpr Options::String help{
+        "The maximum number of times used to calculate the zero crossing of "
+        "the char speeds."};
+    static int lower_bound() { return 3; }
+  };
+
+  using options = tmpl::list<MaxNumTimesForZeroCrossingPredictor>;
+  static constexpr Options::String help{
+      "Computes the control error for size control. Will also write a "
+      "diagnostics file if the control systems are allowed to write data to "
+      "disk."};
+
+  /*!
+   * \brief Initializes the `intrp::ZeroCrossingPredictor`s with the number of
+   * times passed in (default 3).
+   *
+   * \details The internal `control_system::size::Info::state` is initialized to
+   * `control_system::size::States::Initial`. All zero crossing predictors are
+   * initialized with a minimum number of times 3. The input argument
+   * `max_times` represents the maximum number of times.
+   */
+  Size(const int max_times = 3) {
```

Comment:
[optional] Move the non-templated functions into a cpp file.  (The class template argument can be explicitly instantiated.)